server/src/main/java/org/elasticsearch/cluster/service/MasterService.java

```diff
@@ -706,6 +706,7 @@ private static class TaskAckListener {
         public void onCommit(TimeValue commitTime) {
             TimeValue ackTimeout = contextPreservingAckListener.ackTimeout();
             if (ackTimeout == null) {
+                assert false : "ackTimeout must always be present: " + contextPreservingAckListener;
                 ackTimeout = TimeValue.ZERO;
```

Comment:
Should we forbid this when running without assertions as well?

Reply:
I'm inclined to leave the production behaviour alone, for now at least. It's hard to be certain we've covered all the possible `null` values here by static analysis alone.